File: test_server_api.js
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';
import fetch from 'node-fetch';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

// Path to test image
const imagePath = path.join(__dirname, 'public', 'test_hepatic.png');

// Read image as base64
const imageBuffer = fs.readFileSync(imagePath);
const base64Image = imageBuffer.toString('base64');

// Test the server API with both payload formats
async function testServerAPI() {
    console.log('=======================================');
    console.log('TESTING SERVER API ENDPOINT');
    console.log('=======================================');
    console.log(`Image size: ${(imageBuffer.length / 1024).toFixed(2)} KB`);
    
    // First test: Standard format
    await testStandardFormat();
    
    // Second test: b64 format
    await testB64Format();
    
    console.log('=======================================');
    console.log('SERVER API TESTS COMPLETED');
    console.log('=======================================');
}

// Test with standard format (direct base64 string)
async function testStandardFormat() {
    console.log('\n--- Testing with standard format: {content: "base64string"} ---');
    
    const payload = {
        instances: [
            {
                content: base64Image
            }
        ],
        parameters: {
            confidenceThreshold: 0.0,
            maxPredictions: 5
        }
    };
    
    await makeApiRequest(payload, 'standard');
}

// Test with b64 format
async function testB64Format() {
    console.log('\n--- Testing with b64 format: {content: {b64: "base64string"}} ---');
    
    const payload = {
        instances: [
            {
                content: {
                    b64: base64Image
                }
            }
        ],
        parameters: {
            confidenceThreshold: 0.0,
            maxPredictions: 5
        }
    };
    
    await makeApiRequest(payload, 'b64');
}

// Make the API request
async function makeApiRequest(payload, formatType) {
    try {
        console.log(`Sending request to server with ${formatType} format...`);
        
        const start = Date.now();
        const response = await fetch('http://localhost:3003/api/predict', {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(payload)
        });
        const elapsed = Date.now() - start;
        
        console.log(`Response time: ${elapsed}ms`);
        console.log(`Response status: ${response.status} ${response.statusText}`);
        
        if (response.ok) {
            const result = await response.json();
            console.log('Successfully received predictions:');
            
            if (result.displayNames && result.confidences) {
                console.log('Top predictions:');
                result.displayNames.forEach((name, index) => {
                    const confidence = result.confidences[index];
                    console.log(`- ${name}: ${(confidence * 100).toFixed(1)}%`);
                });
            } else {
                console.log('Result structure:', JSON.stringify(result, null, 2));
            }
            
            return true;
        } else {
            let errorText;
            try {
                const errorData = await response.json();
                errorText = JSON.stringify(errorData);
            } catch (e) {
                errorText = await response.text();
            }
            
            console.error(`API error (${formatType} format):`, errorText);
            return false;
        }
    } catch (error) {
        console.error(`Error testing ${formatType} format:`, error.message);
        return false;
    }
}

// Run the tests
testServerAPI(); 

File: test_server_api.py
#!/usr/bin/env python3

import requests
import json
import base64
import os
from PIL import Image
import io

print("========== SERVER API TEST ==========")

# Configuration
SERVER_URL = "http://localhost:3002"
TEST_IMAGE_PATH = "/Users/gabe/VEXUS/test_image.png"
VEIN_TYPE = "hepatic"  # or "portal" or "renal"

# Load the test image
try:
    with open(TEST_IMAGE_PATH, "rb") as image_file:
        image_data = image_file.read()
        image_size = len(image_data)
        print(f"Loaded standard test image, size: {image_size} bytes")

        # Check the actual image type regardless of extension
        img = Image.open(io.BytesIO(image_data))
        mime_type = f"image/{img.format.lower()}"
        print(f"Image format detected: {img.format} (MIME type: {mime_type})")

        # Convert to base64
        base64_data = base64.b64encode(image_data).decode("utf-8")
except FileNotFoundError:
    print(f"Test image not found at {TEST_IMAGE_PATH}. Using fallback.")
    # Create a simple 1x1 pixel PNG as fallback
    img = Image.new('RGB', (1, 1), color = 'white')
    buffered = io.BytesIO()
    img.save(buffered, format="PNG")
    base64_data = base64.b64encode(buffered.getvalue()).decode("utf-8")
    mime_type = "image/png"
    print("Created fallback 1x1 pixel image")

# Create the payload
payload = {
    "instances": [
        {
            "content": base64_data,
            "mimeType": mime_type
        }
    ],
    "parameters": {
        "confidenceThreshold": 0.0,
        "maxPredictions": 5
    },
    "metadata": {
        "veinType": VEIN_TYPE,
        "imageType": mime_type,
        "timestamp": 123456789,
        "onlyOnDemand": False  # Allow fallback to direct calls
    }
}

print(f"\nSending request to server API: {SERVER_URL}/api/predict...")

# Make the request
response = requests.post(
    f"{SERVER_URL}/api/predict", 
    json=payload
)

print(f"Status code: {response.status_code}")

if response.status_code == 200:
    result = response.json()
    print("Server API prediction successful!")
    
    print("\nPredictions:")
    if "displayNames" in result and "confidences" in result:
        for i, (name, conf) in enumerate(zip(result["displayNames"], result["confidences"])):
            if i < 3:  # Just show top 3
                print(f"  {i+1}. {name}: {conf:.2%}")
    else:
        print(json.dumps(result, indent=2))
    
    print(f"\nPrediction method: {result.get('method', 'unknown')}")
else:
    print(f"Server API prediction failed!")
    try:
        error_json = response.json()
        print(f"Error: {json.dumps(error_json, indent=2)}")
    except:
        print(f"Response text: {response.text}")

print("\n========== TEST COMPLETE ==========") 